File: sdk/c#/PicarioXPO.RenderAPI/IXpoUrlFactory.cs
namespace PicarioXPO.RenderAPI
{
    /// <summary>
    /// Factory class for getting the current <c>IXpoUrlGenerator</c> implementation
    /// </summary>
    public interface IXpoUrlFactory
    {
        /// <summary>
        /// Creates a new instance of the currently implemented <c>IXpoUrlGenerator</c> and wraps it
        /// inside the <c>FluentXpoGenerator</c>
        /// </summary>
        IXpoUrlGenerator CreateUrlGenerator();
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/IXpoUrlFactory.ts
 module UrlGeneratorModule {

     export interface IXpoUrlFactory {
         /*
          * Creates a new instance of the currently implemented <c>IXpoUrlGenerator</c> and wraps it
          * inside the <c>FluentXpoGenerator</c>
          */
         createUrlGenerator(): IXpoUrlGenerator ;
     }

 }